ibis/spark/compiler.py

```diff
@@ -0,0 +1,45 @@
+import ibis.sql.compiler as comp
+from ibis.impala.compiler import (
+    ImpalaContext,
+    ImpalaDialect,
+    ImpalaExprTranslator,
+    ImpalaSelect,
+)
+
+
+class SparkSelectBuilder(comp.SelectBuilder):
+    @property
+    def _select_class(self):
+        return SparkSelect
+
+
+class SparkUnion(comp.Union):
+    @staticmethod
+    def keyword(distinct):
+        return 'UNION DISTINCT' if distinct else 'UNION ALL'
```

Comment:
Does Spark SQL actually require the `DISTINCT` part? If not then you don't need to implement this method. This is only for backends that don't follow the usual `UNION`-as-distinct/`UNION ALL`-as-not-distinct pattern. So far the only backend ibis supports that behaves like that is BigQuery.